lightning/src/util/events.rs

```diff
@@ -600,6 +627,20 @@ impl MaybeReadable for Event {
 				};
 				f()
 			},
+			13u8 => {
+				let f = || {
+					let mut payment_id = PaymentId([0; 32]);
+					let mut payment_hash = None;
+					let mut path = VecReadWrapper(Vec::new());
+					read_tlv_fields!(reader, {
+						(0, payment_id, required),
+						(2, payment_hash, option),
+						(4, path, required),
```

Comment:
I think using `vec_type` would clean up the serialization a bit:

```
diff --git a/lightning/src/util/events.rs b/lightning/src/util/events.rs
index 77ad9f75..f47e4793 100644
--- a/lightning/src/util/events.rs
+++ b/lightning/src/util/events.rs
@@ -462,7 +462,7 @@ impl Writeable for Event {
                                write_tlv_fields!(writer, {
                                        (0, payment_id, required),
                                        (2, payment_hash, option),
-                                       (4, VecWriteWrapper(path), required)
+                                       (4, path, vec_type)
                                })
                        },
                        // Note that, going forward, all new events must only write data inside of
@@ -631,13 +631,13 @@ impl MaybeReadable for Event {
                                let f = || {
                                        let mut payment_id = PaymentId([0; 32]);
                                        let mut payment_hash = None;
-                                       let mut path = VecReadWrapper(Vec::new());
+                                       let mut path = None;
                                        read_tlv_fields!(reader, {
                                                (0, payment_id, required),
                                                (2, payment_hash, option),
-                                               (4, path, required),
+                                               (4, path, vec_type),
                                        });
-                                       Ok(Some(Event::PaymentPathSuccessful { payment_id, payment_hash, path: path.0 } ))
+                                       Ok(Some(Event::PaymentPathSuccessful { payment_id, payment_hash, path: path.unwrap() } ))
                                };
                                f()
                        },
```

Reply:
Ah, good point!